### PR TITLE
FreeBSD/clang fixes.

### PR DESCRIPTION
**Pull request by devnexen**

### CODE REVIEW

**Comment by vuvova** [comment]
How do you compile? The change in `storage/rocksdb/build_rocksdb.cmake` is ok, but I don't see why you needed two other changes. No matter how I tried to compile MariaDB 10.2 on FreeBSD with clang, it always succeeded with just `storage/rocksdb/build_rocksdb.cmake` fixed.

**Comment by devnexen** [comment]
FreeBSD 11.1 clang 4.0, basically I get build failures this is how I noticed them, the allocator parameter for the std::map is due to the Allocator::value_type must be same type as value_type assertion failure where indeed the allocator type should match ... the other because all macros from. htobe64 were not defined.

**Comment by vuvova** [comment]
Fair enough. With clang 3.8, `/usr/include/sys/endian` gets included by `/usr/include/c++/v1/__config` which is included by `/usr/include/c++/v1/string`. I can imagine this might not be the case for a different std library version.

File: storage/innobase/include/trx0trx.h [hunk 1]
```diff
@@ -786,7 +786,7 @@ that modified it is running. */
 typedef std::set<
 	dict_table_t*,
 	std::less<dict_table_t*>,
-	ut_allocator<dict_table_t*> >	trx_mod_tables_t;
+	ut_allocator<std::pair<dict_table_t* const, undo_no_t> > >	trx_mod_tables_t;
```
Comment (dr-m) [inline storage/innobase/include/trx0trx.h]: This one looks like it belongs to 10.3, not 10.2. I hope my commit 0c6ff122fa080f80c461f77585176c25c284e33d fixed this.

Comment (devnexen) [reply storage/innobase/include/trx0trx.h]: Should indeed, seems correct to me.

Comment (devnexen) [reply storage/innobase/include/trx0trx.h]: Just pulled it I did not notice your commit lacked the const modifier e.g.
std::pair<dict_table_t* const, undo_no_t> thus still fails :-(

@vuvova what s your opinion ?

**Comment by vuvova** [comment]
freebsd/clang is something we have in buildbot, and also macosx/clang. I don't have an opinion yet, but we'll test the patch and it'll surely work on our builders.

File: storage/innobase/include/trx0trx.h (hunk 1 above)
Comment (grooverdan) [reply storage/innobase/include/trx0trx.h:789]: current 10.5 definition `ut_allocator<std::pair<dict_table_t* const, trx_mod_table_time_t> > >` so it has const now.

File: storage/rocksdb/build_rocksdb.cmake [hunk 2]
```diff
@@ -29,7 +29,6 @@ else()
   endif()
   if(CMAKE_SYSTEM_NAME STREQUAL "FreeBSD")
     # FreeBSD has jemaloc as default malloc
-    add_definitions(-DROCKSDB_JEMALLOC)
```
Comment (grooverdan) [inline storage/rocksdb/build_rocksdb.cmake:32]: is fixed #1599  though was probably fixed before that.

File: storage/rocksdb/rdb_buff.h [hunk 3]
```diff
@@ -40,6 +40,10 @@
 #define be16toh(x) OSSwapBigToHostInt16(x)
 #endif
 
+#ifdef __FreeBSD__
+#include <sys/endian.h>
```
Comment (grooverdan) [inline storage/rocksdb/rdb_buff.h:44]: now part of the rocksdb submodule in port/port_posix.h

**Comment by devnexen** [comment]
> @devnexen, so sorry this was open so long. If its a consolation, all issues you have raised have been fixed.
> 
> Nothing further required. A bunch of other FreeBSD fixes have been merged recently. Despite the slow experience this time, I hope you'll consider submitting again and I'll try get a faster review happening.
> 
> In the meantime, please enjoy MariaDB FreeBSD.

No hard feelings :) I m used to this sort of situations.

**Pull request closed without merging**